### ISSUE TITLE
Syllabus outline

### ISSUE THREAD

**Issue by dill** [pedersen-fisheries-lab/DFO-3day-gam-workshop#1]
I've added a syllabus outline [here](https://github.com/pedersen-fisheries-lab/DFO-3day-gam-workshop/blob/master/syllabus.md). Let me know what you think. The last few sessions I'm a bit unsure on but I think that giving people context can be really helpful. We could also leave things more open-ended.

@eric-pedersen: do you have any info on the research interests of the participants, could we give them a wee google poll?

**Comment by eric-pedersen** [comment]
This syllabus looks great to me! I'm picturing each of these as one 1.5 hour session. 

I think all I'd change is that we can work a lot of the linkages to other software into the the other sections (e.g. DHARMa as a tool for checking models, gratia/mgcViz when we cover plotting GAMs). We could touch on connections to other software at the end of section 5, but I think we can expand section 5 out, and use it as an opportunity to look at individual problems (from my experience, as we get to the third day, the only people left will be the really interested ones). 

We have a pretty good idea of the participant's research interests: about half of them work on stock assessment (measuring population status, rates of change, etc. for different marine fisheries), probably another 1/4 to a third work on marine spatial planning/Species Distribution Modelling/ ecosystem-based management, and there are a mix of other interests (marine mammals, invasive species, data management, pelagic fish, salmon, etc.).

For data: I've been thinking about it, and I think we may want to concentrate on the trawl data set I demonstrated last time we all spoke. It has data on depth, year, lat/long, bottom temperature, and biomass for three key species (cod, shrimp, and turbot). The temperature/depth relationship is a really nice 1-D example that doesn't have the issues to explain that a time series does, and will be immediately understandable by the audience: 

![basic_gam](https://user-images.githubusercontent.com/1554249/88711321-0305f900-d0e6-11ea-8a18-0e04faeb5fd3.png)

It's also easy to fit a multivariate GAM, to add a yearly effect: 

```r
dat = read.csv("data/trawl_nl.csv")
mod = gam(temp_bottom~s(log10(depth))+s(year), data=dat)
gratia::draw(mod)
```
![gam_plot2](https://user-images.githubusercontent.com/1554249/88711514-47919480-d0e6-11ea-84cf-50a228fea3e1.png)

It's also a good test set for spatial and spatio-temporal analyses. 

Finally, the species-specific data are good to test alternative distributions on, as you can use Tweedie for them, or a bionomial distribution to look at presence-absence. 

I've uploaded this to the repo in the data folder. @dill, @gavinsimpson,  and @noamross, could you take a look at it to see if you think it's appropriate for usage for the whole session?

**Comment by dill** [comment]
Glad the syllabus looks okay, happy to have the 5th session be more in-depth. Is your thought there that we would have this be lead by the participats interests and work on-the-fly or that we should prepare more in-depth analyses?

An additional task is to work out who's responsibility it is for each of these bits... I'd really like to get that worked out soon, so I can plan my week next week.

Thanks for the info on the participants, that's very useful to know. Having a group that's a bit more cohesive in their interests should make our lives easier.

I'll take a look at the data, but I think that sounds like a good idea. Do you have a citation for the data source?

**Comment by dill** [comment]
I've just been running through the syllabus with this data and things look interesting. I've committed very rough code [here](https://github.com/pedersen-fisheries-lab/DFO-3day-gam-workshop/blob/master/code_syllabus.R), but some immediate thoughts:

 - the dataset is large, so running the `te(x,y,t)` in a realistic way takes a long time. For that reason I've subsetted to 2010.
 - we have CPUE data (covering `tw`/`nb`/`quasipoisson`/`ZiP`) and the temperature/depth (`gaussian`) but we might be lacking other responses
 - @eric-pedersen do you have a tiling of the strata that could be used for an MRF?
 - I'm wondering if for soap we could put together a simple coastline using `rnaturalearth`?

I'll attempt to finish this Monday, let me know what you think.

**Comment by eric-pedersen** [comment]
I agree that subsetting to 2010 is a good idea. That'll include a good chunk of data, and will still allow 2-d spatial smoothers, te smoothers (e.g. temp_bottom~te(long,lat, depth).

I've added a shape file with all of the strata for the whole shelf (`data/all_strata.shp`) and the strata for just the divisions used in the current data set (`data/trawl_strata.shp`). I've also used `rnaturalearth` to generate a simple coastline for Newfoundland and Labrador (`data/nl_coast.shp`), but the trawls are mostly offshore, so I'm not sure if a soapfilm smoother is the best approach for this data (I don't think restricting the analysis to the boundary would have a huge effect).

For non-normal data: we could use the temperature data to demo the t-distribution, and we can transform biomass into presence-absence easily, so we could use it to demo binomial. I think we could also transform one of the biomass columns into "absence", "low-density", and "commercial density" to demonstrate O-cat. 

Here's a plot with coastline in red, strata in black, and trawls in blue (for all data):

![strata_map](https://user-images.githubusercontent.com/1554249/89108518-2216ca80-d407-11ea-9fdd-b4abbc960d58.png)

**Comment by dill** [comment]
Thanks @eric-pedersen! I've added a soap example in 61a2ef0. Note that the boundary required a bit of fiddling, seems like `rnaturalearth` doesn't necessarily give you sensible boundaries!

**Comment by dill** [comment]
I've also added an MRF example ( 4ba174f) with some absolutely horrendous code because the stratum labels don't match between the data and the shapefile 🙃

**Comment by eric-pedersen** [comment]
I fixed the issues with the strata mismatch (apologies for that... annoying issues with the stratum definition), so I was able to clean up the MRF code a bit: see commit 1cf027f  . 

I've also uploaded a new, cleaned trawl_nl file that I re-created from the database; it now has a column for total (total biomass), shrimp, cod, and richness (i.e. a count of all species found in the trawl). I've scaled the biomasses by area trawled (measured in km^2), but I left `trawl_area` as a variable for use as an offset for modelling richness as a count. Now the dataset should have basically everything we need in one place (so no need to worry about code switching to new data sets). I've also trimmed it to 2005-2014 years, to make it a bit more manageable for model fitting.
